src/SubStatus.h

```diff
@@ -45,21 +48,55 @@
 /// - TransactionSubsMgr::getFullStatus() always returns one of these objects with the std::optional<BlockHeight> as
 ///   the active value, that is, blockHeight() will always be a valid pointer (even if it itself !has_value()).
 ///
+/// - ScriptHashTransactionSubsMgr::getFullStatus() always returns one of these objects with the std::optional<HashSet> as
+///   the active value, that is, hashSet() will always be a valid pointer (even if it itself !has_value()).
+///
+
+using HashSet = std::set<HashX>;
+
+namespace {
+    // assumption: `hashSet` is not empty!
```

Comment:
I guess we can return a magic number `0` for an empty set so that they compare fine